packages/core/src/integrations/rewriteframes.ts

```diff
@@ -1,45 +1,61 @@
+   *
+   * Browser example:
+   * - Original frame: `'http://example.com/my/path/static/asset.js'`
+   * - `root: 'http://example.com/my/path'`
+   * - `assetPrefix: 'app://'`
+   * - Resulting frame: `'app:///static/asset.js'`
+   *
+   * Server example:
+   * - Original frame: `'/User/local/my/path/static/asset.js'`
+   * - `root: '/User/local/my/path'`
+   * - `assetPrefix: 'app://'`
+   * - Resulting frame: `'app:///static/asset.js'`
+   */
   root?: string;
+
+  /**
+   * A custom prefix that stack frames will be prepended with.
+   *
+   * Default: `'app://'`
+   *
+   * This option has slightly different behaviour in the browser and on servers:
+   * - In the browser, the value you provide in `prefix` will prefix the resulting filename when the value you provided in `root` was applied. Effectively replacing whatever `root` matched in the beginning of the frame with `prefix`.
+   * - On the server, the prefix is applied to all stackframes with absolute paths. On Windows, the drive identifier (e.g. "C://") is replaced with the prefix.
+   */
   prefix?: string;
+
+  /**
+   * Defines an iterator that is used to iterate through all of the stack frames for modification before being sent to Sentry.
+   * Setting this option will effectively disable both the `root` and the `prefix` options.
+   */
   iteratee?: StackFrameIteratee;
 }
 
-const _rewriteFramesIntegration = ((options: RewriteFramesOptions = {}) => {
+/**
+ * Rewrite event frames paths.
+ */
+export const rewriteFramesIntegration = defineIntegration((options: RewriteFramesOptions = {}) => {
   const root = options.root;
   const prefix = options.prefix || 'app:///';
 
-  const iteratee: StackFrameIteratee =
-    options.iteratee ||
-    ((frame: StackFrame) => {
-      if (!frame.filename) {
-        return frame;
-      }
-      // Determine if this is a Windows frame by checking for a Windows-style prefix such as `C:\`
-      const isWindowsFrame =
-        /^[a-zA-Z]:\\/.test(frame.filename) ||
-        // or the presence of a backslash without a forward slash (which are not allowed on Windows)
-        (frame.filename.includes('\\') && !frame.filename.includes('/'));
-      // Check if the frame filename begins with `/`
-      const startsWithSlash = /^\//.test(frame.filename);
-      if (isWindowsFrame || startsWithSlash) {
-        const filename = isWindowsFrame
-          ? frame.filename
-              .replace(/^[a-zA-Z]:/, '') // remove Windows-style prefix
-              .replace(/\\/g, '/') // replace all `\\` instances with `/`
-          : frame.filename;
-        const base = root ? relative(root, filename) : basename(filename);
-        frame.filename = `${prefix}${base}`;
-      }
-      return frame;
-    });
+  const isBrowser = 'window' in GLOBAL_OBJ && GLOBAL_OBJ.window !== undefined;
```

Comment:
l: Can we just compress this to `!!GLOBAL_OBJ.window`? I think that should be safe enough..?